File: backend/laravel/app/Http/Controllers/UserController.php
<?php

namespace App\Http\Controllers;

use App\Models\User;
use App\Services\JwtService;
use Illuminate\Http\Request;

class UserController extends Controller
{
    private $jwtService;

    public function __construct(JwtService $jwtService)
    {
        $this->jwtService = $jwtService;
    }

    public function getProfile(Request $request)
    {
        $token = $request->header('Authorization');
        $email = $this->jwtService->getEmailFromToken($token);
        $user = User::where('email', $email)->first();

        return response()->json($user);
    }

    public function updateUsername(Request $request)
    {
        $token = $request->header('Authorization');
        $email = $this->jwtService->getEmailFromToken($token);
        $user = User::where('email', $email)->first();

        $newUsername = $request->input('username');
        $user->username = $newUsername;
        $user->save();

        return response()->json($user);
    }
}

File: backend/flask/app/controllers/user_controller.py
from flask import request, jsonify
from app.services.jwt_service import JwtService
from app.models.user import User

class UserController:
    @staticmethod
    def get_profile():
        token = request.headers.get('Authorization')
        email = JwtService.get_email_from_token(token)
        user = User.query.filter_by(email=email).first()
        return jsonify({'email': user.email, 'username': user.username})

    @staticmethod
    def update_username():
        token = request.headers.get('Authorization')
        email = JwtService.get_email_from_token(token)
        user = User.query.filter_by(email=email).first()

        new_username = request.json.get('username')
        user.username = new_username
        db.session.commit()

        return jsonify({'email': user.email, 'username': user.username})
